spec/controllers/recordings_controller_spec.rb

```diff
@@ -134,13 +135,98 @@
+      post :update_visibility, params: { visibility: 'Unpublished', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Unpublished')
+    end
+
+    it 'changes the recording visibility when changing from Published to Protected locally' do
+      recording = create(:recording, visibility: 'Published')
+      allow_any_instance_of(BigBlueButtonApi).to receive(:update_recordings)
+      post :update_visibility, params: { visibility: 'Protected', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Protected')
+    end
+
+    it 'changes the recording visibility when changing from Unpublished to Protected locally' do
       recording = create(:recording, visibility: 'Unpublished')
-      allow_any_instance_of(BigBlueButtonApi).to receive(:publish_recordings).and_return(http_ok_response)
-      expect { post :publish_recording, params: { publish: 'true', record_id: recording.record_id } }.to change {
-                                                                                                           recording.reload.visibility
-                                                                                                         }.to('Published')
+      allow_any_instance_of(BigBlueButtonApi).to receive(:publish_recordings)
+      allow_any_instance_of(BigBlueButtonApi).to receive(:update_recordings)
+      post :update_visibility, params: { visibility: 'Protected', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Protected')
+    end
+
+    # unpublished -> published
+    it 'changes the recording visibility when changing from Unpublished to Published locally' do
+      recording = create(:recording, visibility: 'Unpublished')
+      allow_any_instance_of(BigBlueButtonApi).to receive(:publish_recordings)
+      post :update_visibility, params: { visibility: 'Published', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Published')
+    end
+
+    # protected -> published
+    it 'changes the recording visibility when changing from Protected to Published locally' do
+      recording = create(:recording, visibility: 'Protected')
+      allow_any_instance_of(BigBlueButtonApi).to receive(:update_recordings)
+      post :update_visibility, params: { visibility: 'Published', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Published')
+    end
+
+    # protected -> unpublished
+    it 'changes the recording visibility when changing from Protected to Unpublished locally' do
+      recording = create(:recording, visibility: 'Protected')
+      allow_any_instance_of(BigBlueButtonApi).to receive(:publish_recordings)
+      allow_any_instance_of(BigBlueButtonApi).to receive(:update_recordings)
+      post :update_visibility, params: { visibility: 'Unpublished', id: recording.record_id }
+      expect(recording.reload.visibility).to eq('Unpublished')
     end
```

Comment:
This is a little overkill since these tests all do the same. I would do just 1 test to make sure the local value is changed